api-gateway-routes/api_gateway_proxy_function.rb

```diff
@@ -41,10 +41,10 @@ def on_connect(event, context)
     :queueUrl => sqs_queue.queue_url,
     :apiEndpoint => api_endpoint,
     :connectionId => request_context["connectionId"],
-    :projectUrl => authorizer["project_url"],
```

Comment:
For my education, where does `authorizer["project_url"]` get set?

Reply:
it gets sent from the client [here](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/javalab/JavabuilderConnection.js#L31) and passed through to javabuilder in [javabuilder_sessions_controller](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/javabuilder_sessions_controller.rb#L43). If we were to update it to send the asset url as well I think we would create that url in javabuilder_sessions_controller.

Reply:
Hmmm.  Not to open a can of worms, but is it ok for us to trust the value that the client sends here?  What happens if the client sends a url to a different project?  What happens if the client sends a random url to another website?  (Feel free to ignore this comment if you don't think it's relevant now.)

Reply:
This is actually solved by this PR because we no longer use it. We could change the pattern to generate the urls on the session token generation side (since we have the same information we are passing to javabuilder there). I'll discuss with the CSA team and decide which route we want to go down.